src/calendar/header/index.tsx

```diff
@@ -235,11 +235,15 @@ const CalendarHeader = forwardRef((props: CalendarHeaderProps, ref) => {
     }
   };
 
+  const renderWeekNumbersSpace = () => {
+    return showWeekNumbers && <View style={style.current.dayHeader}/>;
+  };
+
   const renderDayNames = () => {
     if (!hideDayNames) {
       return (
         <View style={style.current.week} testID={testID ? `${HEADER_DAY_NAMES}-${testID}` : HEADER_DAY_NAMES}>
-          {showWeekNumbers && <Text allowFontScaling={false} style={style.current.dayHeader}></Text>}
+          {renderWeekNumbersSpace()}
           {renderWeekDays}
```

Comment:
Why didn't you migrate here to WeekDaysNames component as well?

Reply:
It is an overlay for the vertical calendar. When you have the week numbers, the calendar offset a day and this means adding the same offset to the weekdays names

Reply:
This header has more functionality, mainly the 'dayTextAtIndex', passing this functionality to WeekDaysNames component will complicate the code as it is adding styles to the header/style StyleSheet

Reply:
IMO it makes more sense to make the WeekDaysNames more generic and maybe a little more complex but support all use cases accordingly. 
Anyway, it's up to you, we can also migrate this use case later on..

Reply:
That was my first thought, and I tried it, but then it became very complex and hard to follow that I decided to leave it for now. We can revisit it in the future.